aidbook/analysis-numail.Rmd

```diff
@@ -0,0 +1,309 @@
+
+# Set node positions (you may want to customize this based on your data)
+pos = nx.spring_layout(G)
+
+# Draw the graph
+plt.figure(figsize=(12, 8))
+nx.draw(G, pos, with_labels=True, node_size=1000, node_color='skyblue', font_size=8, font_color='black', font_weight='bold', edge_color='gray', linewidths=0.5, arrowsize=10)
+plt.title('Aid Inflow Diagram')
+plt.show()
+```
+
+```{python}
+# Create a directed graph
+G = nx.DiGraph()
+
+# Add edges (donor to recipient) based on aid data
+edges = AfricanDB_df[['donor', 'countryname_COW']].dropna().values
+G.add_edges_from(edges)
+
+# Calculate total aid amount received by each country
+total_aid_per_country = df.groupby('countryname_COW')['afdb_projectamount_usd'].sum()
+total_aid_per_country = total_aid_per_country[total_aid_per_country > 0]
+
+# Set node positions (you may want to customize this based on your data)
+pos = nx.spring_layout(G)
+
+# Draw the graph with node sizes based on aid amount
+plt.figure(figsize=(12, 8))
+nx.draw(G, pos, with_labels=True, node_size=[total_aid_per_country.get(country, 1) for country in G.nodes], node_color='skyblue', font_size=8, font_color='black', font_weight='bold', edge_color='gray', linewidths=0.5, arrowsize=10)
+plt.title('Aid Inflow Diagram with Node Size based on Aid Amount')
+plt.show()
+```
+
+```{python}
+# total_aid_per_country.remove()
+i = AfricanDB_df[((AfricanDB_df.countryname_COW != 'Multinational'))]
+i
+
+# AfricanDB_df.drop(i)
+```
+
+```{python}
+total_aid_per_country = i.groupby('countryname_COW')['afdb_projectamount_usd'].sum()
+
+
+total_aid_per_country = total_aid_per_country[total_aid_per_country > 0].sort_values(ascending=False).head(30)
+total_aid_per_country
+
+total_aid_per_country.plot(kind='bar', color='skyblue')
+plt.title('Total Aid per Country')
+plt.xlabel('Country')
+plt.ylabel('Total Aid Amount (USD)')
+plt.show()
+```
+
+```{python}
+# Create a directed graph
+G = nx.DiGraph()
+
+# Add edges (donor to recipient) based on aid data
+edges = AfricanDB_df[['donor', 'countryname_COW']].dropna().values
+G.add_edges_from(edges)
+
+# Calculate total aid amount received by each country
+total_aid_per_country = df.groupby('countryname_COW')['afdb_projectamount_usd'].sum()
+
+# Filter out values less than or equal to 0
+total_aid_per_country = total_aid_per_country[total_aid_per_country > 0]
+
+# Print or inspect total_aid_per_country
+print(total_aid_per_country)
+
+# Set node positions (you may want to customize this based on your data)
+pos = nx.spring_layout(G)
+
+# Print or inspect pos
+print(pos)
+
+# Draw the graph with node sizes based on aid amount
+plt.figure(figsize=(12, 8))
+nx.draw(G, pos, with_labels=True, node_size=[total_aid_per_country.get(country, 1) for country in G.nodes], node_color='skyblue', font_size=8, font_color='black', font_weight='bold', edge_color='gray', linewidths=0.5, arrowsize=10)
+plt.title('Aid Inflow Diagram with Node Size based on Aid Amount (Amount > 0)')
+plt.show()
+
+```
```

Comment:
this doesn't include all of my code, but hopefully after I pull request it should

Reply:
gotcha, once you PR ill add it